Google.PowerShell/Compute/GceHealthCheckCmdlets.cs

```diff
@@ -0,0 +1,545 @@
+                    throw UnknownParameterSetException;
+            }
+        }
+
+        private void RemoveHttpsHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Https Health Check"))
+            {
+                HttpsHealthChecksResource.DeleteRequest request = Service.HttpsHealthChecks.Delete(project, name);
+                Operation operation = request.Execute();
+                AddGlobalOperation(project, operation);
+            }
+        }
+
+        private void RemoveHttpHealthCheck(string project, string name)
+        {
+            if (ShouldProcess($"{project}/{name}", "Remove Http Health Check"))
+            {
+                HttpHealthChecksResource.DeleteRequest request = Service.HttpHealthChecks.Delete(project, name);
+                Operation operation = request.Execute();
+                AddGlobalOperation(project, operation);
+            }
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Sets the data of a Google Compute Engine health check.
+    /// </para>
+    /// <para type="description">
+    /// Sets the data of a Google Compute Engine health check.
+    /// First get the health check object with Get-GceHealthCheck.
+    /// Then change the data in the object you received. Finally send that object to Set-GceHealthCheck.
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Set, "GceHealthCheck", SupportsShouldProcess = true)]
+    [OutputType(typeof(HttpHealthCheck), typeof(HttpsHealthCheck))]
+    public class SetGceHealthCheckCmdlet : GceConcurrentCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByHttpObject = "ByHttpObject";
+            public const string ByHttpsObject = "ByHttpsObject";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The object describing a health check using http.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpHealthCheck HttpObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The object describing a health check using https.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpsHealthCheck HttpsObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            string project;
+            string name;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByHttpObject:
+                    project = GetProjectNameFromUri(HttpObject.SelfLink);
+                    name = HttpObject.Name;
+                    if (ShouldProcess($"{project}/{name}", "Set Http Health Check"))
```

Comment:
For other for `Set-` based cmdlets, I don't recall if we call `ShouldProcess`. Should we be more consistent here? Or are we just doing that here because HTTP health checks are kinda important, unlike tweaking metadata for GCS objects.